### PR TITLE
Allow installing unit tests

### LINKED ISSUE #9343: Install unit tests in new output
**Is your feature request related to a problem? Please describe.**

When cross compiling, one cannot run the unit tests in the current system. It would be very nice to take those unit tests and run them on system we're cross compiling too.

**Describe the solution you'd like**

Create a new output (to not bloat any existing ones) and install the unit tests there.

**Describe alternatives you've considered**

Unit tests can also be run with emulators (Qemu, wine). This is good too! But it shouldn't be seen as a substitute for running the unit tests in the real platform.

**Additional context**

Thankfully, the functional tests and integration tests can already be run separately. But they don't work on Windows yet, while the unit tests do.

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

Closes #9343

See that issue for motivation.

# Context

Installing these is disabled by default, but we enable it (and the additional output we want isntall these too so as not to clutter the existing ones) to use in cross builds and dev shells.

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -167,6 +167,18 @@ AC_ARG_ENABLE(tests, AS_HELP_STRING([--disable-tests],[Do not build the tests]),
   ENABLE_TESTS=$enableval, ENABLE_TESTS=yes)
 AC_SUBST(ENABLE_TESTS)
 
+AC_ARG_ENABLE(install-unit-tests, AS_HELP_STRING([--enable-enstall-unit-tests],[Install the unit tests for running later (default no)]),
```
Comment (knedlsepp) [inline configure.ac:170]: `enstall` -> `install`

Comment (Ericson2314) [reply configure.ac]: Thanks

**Review by thufschmitt — APPROVED** [review]
Looks good

Are you planning on doing the same for the functional tests?

**Comment by Ericson2314** [comment]
@thufschmitt I don't think the bash really needs it since there is nothing to compile? There is the plugin and the standalone executable test, but I think for those *ensuring they compile* is very much part of the test?

At least I've been perfect happy with the "nix against nix" tests since your first introduced them in a0866c8ea4bc66f9aacc7ad19139d57946b3df18 :)

**Pull request merged**